File: esp32/main.py
"""Main entry point for Braille Label Maker (MicroPython ESP32)"""

import time

# Initialize all components
from components import (
    wifi_manager, solenoids, roller_motor, vibration,
    audio, camera_module, power
)

from handlers import (
    print_handler, feed_handler, camera_handler,
    voice_to_braille, camera_to_braille, web_handler
)

from test_components import startup_tests


def main():
    """Main entry point"""
    
    # Wait a bit for the serial monitor to catch up
    time.sleep(1)
    
    print("\n" + "=" * 50)
    print("Braille Label Maker (MicroPython)")
    print("=" * 50 + "\n")
    
    # Initialize WiFi
    wifi_manager.init_wifi()
    
    # Initialize components
    solenoids.init()
    roller_motor.init()
    vibration.init()
    power.init()
    
    # Initialize handlers
    print_handler.init()
    feed_handler.init()
    camera_handler.init()
    voice_to_braille.init()
    camera_to_braille.init()
    web_handler.init()
    
    # Run startup tests
    startup_tests.run_all()
    
    # Start web server (this blocks)
    print("Starting web server...")
    web_handler.start_web_server(port=80)


if __name__ == "__main__":
    try:
        main()
    except KeyboardInterrupt:
        print("\nShutdown requested")
    except Exception as e:
        print(f"\nFatal error: {e}")
        import traceback
        traceback.print_exc()

File: esp32/src/main.cpp
#include <Arduino.h>
#include <WiFi.h>
#include <WebServer.h>
#include <secrets.h>

WebServer server(80);

#include "../handlers/web_handler.h"

void setup() {
  Serial.begin(115200);
  delay(300);
  Serial.println();
  Serial.println("Braille label maker — starting");

  WiFi.mode(WIFI_STA);
  WiFi.begin(WIFI_SSID, WIFI_PASSWORD);

  Serial.print("WiFi connecting to ");
  Serial.println(WIFI_SSID);
  uint8_t tries = 0;
  while (WiFi.status() != WL_CONNECTED && tries < 80) {
    delay(250);
    Serial.print(".");
    tries++;
  }
  Serial.println();

  if (WiFi.status() == WL_CONNECTED) {
    Serial.print("WiFi connected. IP: ");
    Serial.println(WiFi.localIP());
  } else {
    Serial.println("WiFi connection failed — check secrets.h SSID/password.");
  }

  PrintHandler::init();
  WebHandler::init();
}

void loop() {
  server.handleClient();
}
